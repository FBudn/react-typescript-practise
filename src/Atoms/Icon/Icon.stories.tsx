import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Icon from "./Icon";

export default {
  title: "Atoms/Icon",
  component: Icon,
  argTypes: {
    onClickIcon: { action: "Console.log: `My name Clicked` " },
  },
} as ComponentMeta<typeof Icon>;

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} />;

export const BasicIcon = Template.bind({});
