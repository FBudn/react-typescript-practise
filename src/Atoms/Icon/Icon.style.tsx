import styled from "styled-components";

export const SoloIconContainer = styled.div.attrs(
  (props: { color: string }) => props,
)`
  display: flex;
  width: 25px;
  height: 25px;
  justify-content: center;
  align-items: center;
  padding: 8px;
  font-size: 1rem;
  border-radius: 100%;
  border: 2px solid red;
  cursor: pointer;

  border-color: ${(props) => props.color};
  color: ${(props) => props.color};
`;
