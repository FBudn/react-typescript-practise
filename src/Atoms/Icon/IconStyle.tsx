import styled from "styled-components";

export const SoloIconContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  text: center;
  padding: 8px;
  font-size: 1rem;
  border-radius: 100%;
  border: 2px solid black;
  cursor: pointer;
`;
