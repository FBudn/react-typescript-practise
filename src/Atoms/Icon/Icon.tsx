import {
  faFacebookF,
  faGoogle,
  faLinkedinIn,
} from "@fortawesome/free-brands-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import React from "react";
import { SoloIconContainer } from "./Icon.style";

export interface IconProps {
  onClickIcon: (i: string) => void;
  color: string;
  icon: "facebook" | "google" | "linkedin";
}

const Icon: React.FC<IconProps> = ({ onClickIcon, color, icon }) => {
  switch (icon) {
    case "facebook":
      return (
        <SoloIconContainer
          data-testid="Icon-1-case"
          color={color}
          onClick={() => {
            onClickIcon(`Facebook Clicked`);
          }}
        >
          <FontAwesomeIcon icon={faFacebookF} />
        </SoloIconContainer>
      );
    case "google":
      return (
        <SoloIconContainer
          data-testid="Icon-2-case"
          color={color}
          onClick={() => {
            onClickIcon(`Google Clicked`);
          }}
        >
          <FontAwesomeIcon icon={faGoogle} />
        </SoloIconContainer>
      );
    case "linkedin":
      return (
        <SoloIconContainer
          data-testid="Icon-3-case"
          color={color}
          onClick={() => {
            onClickIcon(`LinkedIn Clicked`);
          }}
        >
          <FontAwesomeIcon icon={faLinkedinIn} />
        </SoloIconContainer>
      );

    default:
      return (
        <SoloIconContainer
          color={color}
          onClick={() => {
            onClickIcon(`Facebook Clicked`);
          }}
        >
          <FontAwesomeIcon icon={faFacebookF} />
        </SoloIconContainer>
      );
  }
};

export default Icon;
