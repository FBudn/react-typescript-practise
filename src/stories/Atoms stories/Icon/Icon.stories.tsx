import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFacebookF } from "@fortawesome/free-brands-svg-icons";
import { string } from "prop-types";
import Icon from "../../../Atoms/Icon/Icon";

export default {
  title: "Atoms/Icon",
  component: Icon,
  argTypes: {
    onClickIcon: { action: "Console.log: `My name Clicked` " },
    color: string,
  },
  decorators: [
    (Story) => (
      <div style={{ backgroundColor: "transparent", width: "30px" }}>
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof Icon>;

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} />;

export const BasicIcon = Template.bind({});
BasicIcon.args = {
  children: <FontAwesomeIcon icon={faFacebookF} />,
};

// enum przetrenowaÄ‡
