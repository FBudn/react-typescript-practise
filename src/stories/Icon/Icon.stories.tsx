/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFacebookF } from "@fortawesome/free-brands-svg-icons";
import { any, string } from "prop-types";
import Icon from "../../Atoms/Icon/Icon";

export default {
  title: "Atoms/Icon",
  component: Icon,
  argTypes: {
    onClickIcon: { action: "Console.log: `My name Clicked` " },
    color: string,
    children: any,
  },
};
const facebook = <FontAwesomeIcon icon={faFacebookF} />;

const Template = (args: any) => <Icon {...args}>{facebook}</Icon>;

export const BasicIcon = Template.bind({});
