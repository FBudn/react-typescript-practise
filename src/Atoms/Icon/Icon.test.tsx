import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Icon from "./Icon";
import "jest-styled-components";

const mockOnClickIcon = jest.fn();

test(`Should render Icon component`, () => {
  render(<Icon onClickIcon={mockOnClickIcon} color="red" icon="facebook" />);
  render(<Icon onClickIcon={mockOnClickIcon} color="blue" icon="google" />);
  render(<Icon onClickIcon={mockOnClickIcon} color="blue" icon="linkedin" />);
  const IconElement1 = screen.getByTestId(`Icon-1-case`);
  const IconElement2 = screen.getByTestId(`Icon-2-case`);
  const IconElement3 = screen.getByTestId(`Icon-3-case`);
  expect(IconElement1).toBeInTheDocument();
  expect(IconElement2).toBeInTheDocument();
  expect(IconElement3).toBeInTheDocument();
});

test(`Should match snapshot and have styles:`, () => {
  const IconElement1 = TestRenderer.create(
    <Icon onClickIcon={mockOnClickIcon} color="red" icon="facebook" />,
  ).toJSON();
  const IconElement2 = TestRenderer.create(
    <Icon onClickIcon={mockOnClickIcon} color="blue" icon="google" />,
  ).toJSON();
  const IconElement3 = TestRenderer.create(
    <Icon onClickIcon={mockOnClickIcon} color="blue" icon="linkedin" />,
  ).toJSON();

  expect(IconElement1).toMatchSnapshot();
  expect(IconElement2).toMatchSnapshot();
  expect(IconElement3).toMatchSnapshot();
});

test(`Should render have styles:`, () => {
  const IconElement1 = TestRenderer.create(
    <Icon onClickIcon={mockOnClickIcon} color="red" icon="facebook" />,
  ).toJSON();
  const IconElement2 = TestRenderer.create(
    <Icon onClickIcon={mockOnClickIcon} color="blue" icon="google" />,
  ).toJSON();
  const IconElement3 = TestRenderer.create(
    <Icon onClickIcon={mockOnClickIcon} color="blue" icon="linkedin" />,
  ).toJSON();

  expect(IconElement1).toHaveStyleRule("display", "flex");
  expect(IconElement1).toHaveStyleRule("width", "25px");
  expect(IconElement1).toHaveStyleRule("height", "25px");
  expect(IconElement1).toHaveStyleRule("justify-content", "center");
  expect(IconElement1).toHaveStyleRule("align-items", "center");
  expect(IconElement1).toHaveStyleRule("padding", "8px");
  expect(IconElement1).toHaveStyleRule("font-size", "1rem");
  expect(IconElement1).toHaveStyleRule("border-radius", "100%");
  expect(IconElement1).toHaveStyleRule("cursor", "pointer");
  expect(IconElement2).toHaveStyleRule("display", "flex");
  expect(IconElement2).toHaveStyleRule("width", "25px");
  expect(IconElement2).toHaveStyleRule("height", "25px");
  expect(IconElement2).toHaveStyleRule("justify-content", "center");
  expect(IconElement2).toHaveStyleRule("align-items", "center");
  expect(IconElement2).toHaveStyleRule("padding", "8px");
  expect(IconElement2).toHaveStyleRule("font-size", "1rem");
  expect(IconElement2).toHaveStyleRule("border-radius", "100%");
  expect(IconElement2).toHaveStyleRule("cursor", "pointer");
  expect(IconElement3).toHaveStyleRule("display", "flex");
  expect(IconElement3).toHaveStyleRule("width", "25px");
  expect(IconElement3).toHaveStyleRule("height", "25px");
  expect(IconElement3).toHaveStyleRule("justify-content", "center");
  expect(IconElement3).toHaveStyleRule("align-items", "center");
  expect(IconElement3).toHaveStyleRule("padding", "8px");
  expect(IconElement3).toHaveStyleRule("font-size", "1rem");
  expect(IconElement3).toHaveStyleRule("border-radius", "100%");
  expect(IconElement3).toHaveStyleRule("cursor", "pointer");
});

test(`should handle onClicks`, () => {
  const mockOnClickIcon1 = jest.fn();
  const mockOnClickIcon2 = jest.fn();
  const mockOnClickIcon3 = jest.fn();

  render(<Icon onClickIcon={mockOnClickIcon1} color="red" icon="facebook" />);
  const consoleLogIcon1 = screen.getByTestId(`Icon-1-case`);
  fireEvent.click(consoleLogIcon1);
  expect(mockOnClickIcon1).toBeCalledTimes(1);
  mockOnClickIcon1(`Icon Test 1`);
  expect(mockOnClickIcon1).toBeCalledWith(`Icon Test 1`);

  render(<Icon onClickIcon={mockOnClickIcon2} color="blue" icon="google" />);
  const consoleLogIcon2 = screen.getByTestId(`Icon-2-case`);
  fireEvent.click(consoleLogIcon2);
  expect(mockOnClickIcon2).toBeCalledTimes(1);
  mockOnClickIcon2(`Icon Test 2`);
  expect(mockOnClickIcon2).toBeCalledWith(`Icon Test 2`);

  render(<Icon onClickIcon={mockOnClickIcon3} color="blue" icon="linkedin" />);
  const consoleLogIcon3 = screen.getByTestId(`Icon-3-case`);
  fireEvent.click(consoleLogIcon3);
  expect(mockOnClickIcon3).toBeCalledTimes(1);
  mockOnClickIcon3(`Icon Test 3`);
  expect(mockOnClickIcon3).toBeCalledWith(`Icon Test 3`);
});
