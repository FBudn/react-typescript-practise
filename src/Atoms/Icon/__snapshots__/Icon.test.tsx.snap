// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot and have styles: 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 25px;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  font-size: 1rem;
  border-radius: 100%;
  border: 2px solid red;
  cursor: pointer;
  border-color: red;
  color: red;
}

<div
  className="c0"
  color="red"
  data-testid="Icon-1-case"
  onClick={[Function]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-facebook-f "
    data-icon="facebook-f"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`Should match snapshot and have styles: 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 25px;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  font-size: 1rem;
  border-radius: 100%;
  border: 2px solid red;
  cursor: pointer;
  border-color: blue;
  color: blue;
}

<div
  className="c0"
  color="blue"
  data-testid="Icon-2-case"
  onClick={[Function]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-google "
    data-icon="google"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 488 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`Should match snapshot and have styles: 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 25px;
  height: 25px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  font-size: 1rem;
  border-radius: 100%;
  border: 2px solid red;
  cursor: pointer;
  border-color: blue;
  color: blue;
}

<div
  className="c0"
  color="blue"
  data-testid="Icon-3-case"
  onClick={[Function]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-linkedin-in "
    data-icon="linkedin-in"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
